src/main/java/mate/academy/security/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,48 @@
+package mate.academy.security.impl;
+
+import java.util.Optional;
+import mate.academy.dao.UserDao;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.security.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserDao userDao;
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDb = userService.findByEmail(email);
+        if (userFromDb.isEmpty()) {
+            throw new AuthenticationException("Can`t login a user by empty email");
+        }
+        User user = userFromDb.get();
+        String hashPassword = HashUtil.hashPassword(password, user.getSalt());
+        if (user.getPassword().equals(hashPassword)) {
+            return user;
+        }
+        throw new AuthenticationException("Can`t login a user by email(" + email + ")."
+                + " Wrong password.");
```

Comment:
use formatting for exception if you want add email value and don`t write that only password is wrong!